### PR TITLE
[libxml2] Use official CMakeLists and add feature tools

### LINKED ISSUE #15311: [libxml2] Build the tools: xmllint, xmlcatalog
libxml2 has tools: `xmllint`, `xmlcatalog`
Let's build them and install to `tools/libxml2`

### LINKED ISSUE #16298: [libxml2] libxml2:x64-windows-static is not statically linked with libiconv
****

**Is your feature request related to a problem? Please describe.**

When linking `libxml:x64-windows-static` it produces errors like these, because it is not statically linked with `libiconv`:
```
lld-link: error: undefined symbol: libiconv_open
>>> referenced by E:\git\vcpkg\buildtrees\libxml2\src\v2.9.10-073f92c335.clean\encoding.c:1691
>>>               libxml2.lib(encoding.c.obj):($LN82)
>>> referenced by E:\git\vcpkg\buildtrees\libxml2\src\v2.9.10-073f92c335.clean\encoding.c:1692
>>>               libxml2.lib(encoding.c.obj):($LN82)
>>> referenced by E:\git\vcpkg\buildtrees\libxml2\src\v2.9.10-073f92c335.clean\encoding.c:1694
>>>               libxml2.lib(encoding.c.obj):($LN82)
>>> referenced 1 more times

lld-link: error: undefined symbol: libiconv
>>> referenced by E:\git\vcpkg\buildtrees\libxml2\src\v2.9.10-073f92c335.clean\encoding.c:2827
>>>               libxml2.lib(encoding.c.obj):($LN45)
>>> referenced by E:\git\vcpkg\buildtrees\libxml2\src\v2.9.10-073f92c335.clean\encoding.c:2611
>>>               libxml2.lib(encoding.c.obj):($retry$111)
>>> referenced by E:\git\vcpkg\buildtrees\libxml2\src\v2.9.10-073f92c335.clean\encoding.c:2684
>>>               libxml2.lib(encoding.c.obj):($retry$111)
>>> referenced 9 more times

lld-link: error: undefined symbol: libiconv_close
>>> referenced by E:\git\vcpkg\buildtrees\libxml2\src\v2.9.10-073f92c335.clean\encoding.c:1703
>>>               libxml2.lib(encoding.c.obj):($LN82)
>>> referenced by E:\git\vcpkg\buildtrees\libxml2\src\v2.9.10-073f92c335.clean\encoding.c:1704
>>>               libxml2.lib(encoding.c.obj):($LN82)
>>> referenced by E:\git\vcpkg\buildtrees\libxml2\src\v2.9.10-073f92c335.clean\encoding.c:2744
>>>               libxml2.lib(encoding.c.obj):($LN29)
>>> referenced 1 more times
```

**Proposed solution**

Statically link to `libiconv` when building `libxml2:x64-windows-static`

**Describe alternatives you've considered**

Could be resolved by #15390

**Additional context**

### PR DESCRIPTION
**Pull request by JackBoosY**
Libxml2 officially supports the cmake build method, but the previous version has not been released for a year. Use official CMakeLists.

Fixes #15311 #14726.

### CODE REVIEW

File: ports/libxml2/fix-version-define.patch [hunk 1]
```diff
@@ -0,0 +1,10 @@
+diff --git a/win32/libxml2.rc b/win32/libxml2.rc
+index 08e2da1..c774a5e 100644
+--- a/win32/libxml2.rc
++++ b/win32/libxml2.rc
+@@ -1,4 +1,5 @@
+ #include <winver.h>
++#include "rcVersion.h"
```
Comment (JackBoosY) [inline ports/libxml2/fix-version-define.patch:7]: This changes has been merged to the upstream master:
https://github.com/GNOME/libxml2/blob/master/win32/libxml2.rc

**Comment by clemenswasser** [comment]
Hi @JackBoosY.
what is the current status? I have tried using this branch, but sadly it does not work.
Is it blocked by upstream?

**Comment by JackBoosY** [comment]
@clemenswasser  I don't have enough time yet, will restart this job later.

**Comment by clemenswasser** [comment]
@JackBoosY Awesome Work! 🥳
This already partially works for me.
Although I have some problems with Iconv (which is a dependency of LibXml2)
```
CMake Error at <VCPKG_ROOT>/installed/x64-windows-static/share/iconv/vcpkg-cmake-wrapper.cmake:12 (target_link_libraries):
  Cannot specify link libraries for target "Iconv::Iconv" which is not built
  by this project.
```

**Comment by JackBoosY** [comment]
@clemenswasser  You should remove `FindICONV.cmake` in _SOURCE/cmake_

**Comment by clemenswasser** [comment]
You mean `C:\Program Files\CMake\share\cmake-3.20\Modules\FindIconv.cmake`?

**Comment by JackBoosY** [comment]
@clemenswasser  No, on which port did you encounter the problem?

**Comment by clemenswasser** [comment]
Not on a port. I was building llvm from source.
The full stack trace:
```
CMake Error at <VCPKG_ROOT>/installed/x64-windows-static/share/iconv/vcpkg-cmake-wrapper.cmake:12 (target_link_libraries):
  Cannot specify link libraries for target "Iconv::Iconv" which is not built
  by this project.
Call Stack (most recent call first):
  <VCPKG_ROOT>/scripts/buildsystems/vcpkg.cmake:815 (include)
  C:/Program Files/CMake/share/cmake-3.20/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  <VCPKG_ROOT>/installed/x64-windows-static/share/libxml2/libxml2-config.cmake:107 (find_dependency)
  <VCPKG_ROOT>/installed/x64-windows-static/share/libxml2/vcpkg-cmake-wrapper.cmake:3 (_find_package)
  <VCPKG_ROOT>/scripts/buildsystems/vcpkg.cmake:815 (include)
  <LLVM_ROOT>/lldb/cmake/modules/LLDBConfig.cmake:48 (find_package)
  <LLVM_ROOT>/lldb/cmake/modules/LLDBConfig.cmake:60 (add_optional_dependency)
  <LLVM_ROOT>/lldb/CMakeLists.txt:21 (include)
```
Which tells me that the `iconv/vcpkg-cmake-wrapper.cmake` is wrong.

**Comment by JackBoosY** [comment]
Only have libiconv issue: https://gitlab.kitware.com/cmake/cmake/-/issues/22114#note_945535.

**Comment by clemenswasser** [comment]
> Only have libiconv issue: https://gitlab.kitware.com/cmake/cmake/-/issues/22114.

So using libxml2 (which depends on libiconv) as a static library works for you?
I will look into the error that I posted. I think that it's an issue of your libiconv port's `vcpkg-cmake-wrapper.cmake`. It tries to add the charset libraries to the `Iconv::Iconv` target, but this is not allowed in cmake. This was introduced in https://github.com/microsoft/vcpkg/commit/3a26eb3dc4d684ca6c1748dccc7a15684b9e8d0a

**Comment by JackBoosY** [comment]
@clemenswasser No,
https://github.com/microsoft/vcpkg/blob/84e5852cfaa8e7875027e79a29a5534bb5d4a475/ports/libiconv/vcpkg-cmake-wrapper.cmake#L4
This conditional judgment did not enter, I confirmed.

**Comment by JackBoosY** [comment]
The upstream was fixed this issue, waiting for update cmake to 3.21.0.

**Comment by clemenswasser** [comment]
Hi @JackBoosY,
after playing a bit more with your changes I realized that everything is working correctly and that there is only a little issue, because of which it didn't work for me. The problem is that the current cmake config files don't define `LIBXML2_FOUND` (which [llvm relies on](https://github.com/llvm/llvm-project/blob/6974f1843321ee4d15b7d82d0ccfb4e8cbe76f0b/clang/CMakeLists.txt#L208)). Can you add this on your side or is this an upstream bug?

(Thanks for your awesome work 😀)

**Comment by JackBoosY** [comment]
@clemenswasser I prefer to use PR https://github.com/microsoft/vcpkg/pull/17945 instead of this.

**Pull request closed without merging**